pkg/api/vlabs/types.go

```diff
@@ -42,6 +42,7 @@ type Properties struct {
 	CertificateProfile      *CertificateProfile      `json:"certificateProfile,omitempty"`
 	AADProfile              *AADProfile              `json:"aadProfile,omitempty"`
 	AzProfile               *AzProfile               `json:"azProfile,omitempty"`
+	Debug                   *Debug                   `json:"debug,omitempty"`
```

Comment:
thoughts on calling this "ExperimentalFeatures" or "FeatureFlags" instead? To me "debug" sounds like a bool that can turn on or off more debugging output

Reply:
I made this same point offline. I thought of "Experimental" or "Incubator" (or "Danger") but I like @CecileRobertMichon's suggestions better.

Reply:
How about TimeTrial?

Reply:
Or RunAsync

Reply:
IsRunInBackgroundEnabled?

Reply:
isn't this new field supposed to be a general "new features to include" struct? `TimeTrial` and `RunAsync` sound directly related to CSE async provisioning

Reply:
https://github.com/Azure/acs-engine/pull/4104/files/174b85b7870648b1abeaf550638844df360cbd94#diff-0226d5bd49c7195fda1784e5dd4274c9R58 is the specific one

Reply:
ok on a serious note, DebugSettings?

Reply:
CSEDebugSettings?

Reply:
How about `Proto`? This is not meant to be just for "features". It is meant to represent  operational break glass scenarios, or "extra" tooling for testing/debugging.

Here are some examples:

- deliver load/performance workloads
- deliver long-running introspection (e.g., tcpdump)
- alternate/experimental implementations for A/B testing